src/wasm/wasm-binary.cpp

```diff
@@ -1549,8 +1549,28 @@ void WasmBinaryWriter::trackExpressionDelimiter(Expression* curr,
 }
 
 std::optional<BufferWithRandomAccess> WasmBinaryWriter::writeCodeAnnotations() {
-  // Assemble the info for Branch Hinting: for each function, a vector of the
-  // hints.
+  std::optional<BufferWithRandomAccess> ret;
+
+  auto append = [&](std::optional<BufferWithRandomAccess>&& temp) {
+    if (temp) {
+      if (!ret) {
+        // This is the first section.
+        ret = std::move(temp);
+      } else {
+        // This is a later section, append.
+        ret->insert(ret->end(), temp->begin(), temp->end());
+      }
+    }
+  };
+
+  append(writeBranchHints());
```

Comment:
I think this would be clearer if `writeBranchHints` were named `getBranchHintBuffer` or similar.

Reply:
Sounds good, renamed.